.github/workflows/ci.yml

```diff
@@ -220,6 +221,27 @@ jobs:
           make VERBOSE=1 -j${BUILD_CONCURRENCY} ${make_targets}
           ctest --timeout ${TEST_TIMEOUT} --output-on-failure ${ctest_flags}
 
+  dr-testing:
+    name: Run DR shp-tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build docker image
+        run: docker build . -t dr-test -f .github/Dockerfile
```

Comment:
Can we cache the docker image if we really need a container? This would speed up the overall CI. It looks like the current DR test setup takes significantly longer than the other CI jobs.

Reply:
After removing docker and switching to oneDPL way of doing CI the job is fast, e.g. [this execution](https://github.com/oneapi-src/oneDPL/actions/runs/9857976656/job/27218248103?pr=1663) took 7 minutes, which is not longer than oneDPL tests average.